File: ycombinator_easier.js
#!/usr/bin/env node
Y = function (builder, n) { return builder(builder, n); };

fact_helper = function (builder, n) { if (n == 0) return 1; else return n * builder(builder, n - 1); };

fact = function(n) { return Y(fact_helper, n); };

console.log(fact(21));

File: ycombinator_easier.py
#!/usr/bin/env python2
from functools import partial

Y = lambda builder,n: builder(builder,n)

fact_helper = lambda builder,n: 1 if n == 0 else n*builder(builder,n-1)

fact = partial(Y, fact_helper)

print "21! =", fact(21)
